File: framework/languages/dotnet/samples/ShoppingMall/Server/OrderWorkflow/Adapters/ZLink/Spots/Handlers/ContinueOrderWorkflowHandler.cs
using ShoppingMall.Server.OrderWorkflow.Adapters.ZLink.Spots;
using ShoppingMall.Shared.Contracts;
using Zlink.Framework.Contracts.Spots;

namespace ShoppingMall.Server.OrderWorkflow.Adapters.ZLink.Spots.Handlers;

internal sealed class ContinueOrderWorkflowHandler :
    IZLinkSpotRequestHandler<OrderWorkflowSpot, ContinueOrderWorkflowReq, ContinueOrderWorkflowRes>
{
    public ValueTask<ContinueOrderWorkflowRes> HandleAsync(
        OrderWorkflowSpot spot,
        ContinueOrderWorkflowReq request,
        CancellationToken cancellationToken)
    {
        return spot.ContinueOrderWorkflowAsync(request, cancellationToken);
    }
}

File: framework/languages/node/samples/ShoppingMall.Ts/Server/OrderWorkflow/Handlers/continue-order-workflow-handler.ts
import { Inject } from '@nestjs/common';
import { ZLINK_SPOT_MANAGER } from '@zlink-systems/nestjs';
import { OrderWorkflowService } from '../Application/OrderWorkflow/order-workflow-service';
import { ensureOrderWorkflowSpot } from './start-order-handler';
import type { ZLinkRouteRequestContext, ZLinkRouteRequestHandler, ZLinkSpotManager } from '@zlink-systems/framework';
import type { ContinueOrderWorkflowReq, ContinueOrderWorkflowRes } from '../../../Shared/Contracts/messages';

class ContinueOrderWorkflowHandler implements ZLinkRouteRequestHandler<ContinueOrderWorkflowReq, ContinueOrderWorkflowRes> {
  constructor(
    @Inject(OrderWorkflowService) private readonly workflow: OrderWorkflowService,
    @Inject(ZLINK_SPOT_MANAGER) private readonly spots: ZLinkSpotManager
  ) {}

  async handle(request: ContinueOrderWorkflowReq, context: ZLinkRouteRequestContext): Promise<ContinueOrderWorkflowRes> {
    void context;
    await ensureOrderWorkflowSpot(this.spots, request.orderId);
    return this.workflow.continue(request);
  }
}

export {
  ContinueOrderWorkflowHandler
};
